packages/rocketchat-lib/server/methods/sendMessage.js

```diff
@@ -62,13 +62,37 @@ Meteor.methods({
 			throw new Meteor.Error('You can\'t send messages because you are blocked');
 		}
 
+		if (room.ro === true) {
+			const userOwner = RocketChat.models.Roles.findOne({
```

Comment:
same as above. I don't think we should test room's owner. Since `post-readonly` is given by default to _owners_ and _moderators_ I think we can test only the permission.